pyuploadcare/api_resources.py

```diff
@@ -640,36 +631,66 @@ def __init__(self, **kwargs):
             setattr(self, f, kwargs.pop(f, default))
 
         if kwargs:
-            raise TypeError('Extra arguments: {}.'.format(
+            raise TypeError('Extra arguments: {0}.'.format(
                 ", ".join(kwargs.keys())
             ))
 
     def api_url(self, **additional):
         qs = {}
-        if self.since is not None:
-            qs['from'] = self.since.isoformat()
-        if self.until is not None:
-            qs['to'] = self.until.isoformat()
+
+        if self.starting_point is not None:
+            qs['from'] = self.starting_point
+
+        if self.ordering is not None:
+            qs['ordering'] = self.ordering
+
         if self.request_limit:
             qs['limit'] = self.request_limit
+
         for f, default in self.filters:
             v = getattr(self, f)
             if v == default:
                 continue
             qs[f] = 'none' if v is None else str(bool(v)).lower()
+
         qs.update(additional)
+
         return self.base_url + '?' + urlencode(qs)
 
+    @property
+    def starting_point(self):
```

Comment:
There is no need for such complicated logic. Just a dictionary should be enough:

```python
from_converters = {
  None: datetime.isoformat,
  'datetime_uploaded': datetime.isoformat,
  'size': str,
}
...

if self.from is not None:
  from_converter = self.from_converters[self.ordering.lstrip('-')]
  qs['from'] = from_converter(self.starting_point)
```